torchgeo/datasets/stacapidataset.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""STACAPIDataset."""
+
+import abc
+import sys
+import time
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, cast
+
+import fiona
+import matplotlib.pyplot as plt
+import numpy as np
+import planetary_computer as pc
+import stackstac
+import torch
+from pystac.item import Item
+from pystac_client import Client
+from rasterio.crs import CRS
+from rioxarray.merge import merge_arrays
+from rtree.index import Index, Property
+from torch import Tensor
+
+# from torch.utils.data import DataLoader
+from xarray.core.dataarray import DataArray
+
+from torchgeo.datasets.geo import RasterDataset
+from torchgeo.datasets.utils import BoundingBox
+
+# from torchgeo.samplers import RandomGeoSampler
+
+
+class STACAPIDataset(RasterDataset, abc.ABC):
```

Comment:
```suggestion
class STACAPIDataset(GeoDataset):
```
This is neither an ABC (there are no abstract methods), nor does it need to be a subclass of `RasterDataset` (it doesn't re-use any of the logic from `RasterDataset`). I would consider this to be just another possible base class that subclasses `GeoDataset`.